opencda/scenario_testing/utils/sim_api.py

```diff
@@ -536,14 +597,21 @@ def spawn_vehicle_by_range(self, tm, traffic_config, bg_list):
                                                   yaw=coordinates[4],
                                                   pitch=coordinates[5]))
             if not traffic_config['random']:
-                ego_vehicle_bp.set_attribute('color', '0, 255, 0')
+                ego_vehicle_bp.set_attribute('color', color)
 
             else:
+                # sample a bp from various classes
+                if self.use_multi_class_bp:
+                    label = np.random.choice(label_list, p=prob)
+                    # Given the label (class), find all associated blueprints in CARLA
+                    ego_vehicle_random_list = multi_class_vehicle_blueprint_filter(
```

Comment:
good code here,  use the `ego_vehicle_random_list` for both single class or multi_class_bp